web_external/js/views/body/UploadDatasetMetadataView.js

```diff
@@ -0,0 +1,339 @@
+            this.uploadFolder = new girder.models.FolderModel({
+                name: 'isic_upload_' + Date.now(),
+                parentType: 'user',
+                parentId: girder.currentUser.get('_id'),
+                description: 'ISIC uploads'
+            });
+
+            this.uploadFolder.once('g:saved', function () {
+                this.startUpload(this.uploadFolder);
+            }, this).once('g:error', function () {
+                girder.events.trigger('g:alert', {
+                    icon: 'cancel',
+                    text: 'Could not create upload folder.',
+                    type: 'error',
+                    timeout: 4000
+                });
+            }, this).save();
+        }
+    },
+
+    uploadFinished: function (files) {
+        this.uploadedCsvFiles = _.pluck(files.files, 'name');
+        this.updateUploadWidget();
+    },
+
+    startUpload: function (folder) {
+        // Configure upload widget and begin upload
+        this.uploadWidget.parentType = 'folder';
+        this.uploadWidget.parent = folder;
+        this.uploadWidget.uploadNextFile();
+    },
+
+    updateUploadWidget: function () {
+        var visible = false;
+        var uploadList = [];
+        if (this.uploadedCsvFiles.length) {
+            visible = false;
+            uploadList = this.uploadedCsvFiles;
+        } else {
+            visible = true;
+        }
+
+        this.$('.isic-upload-widget-container').toggle(visible);
+        this.$('.isic-upload-reset-container').toggle(!visible);
+
+        this.uploadWidget.render();
+        this.$('.isic-upload-list').text(
+            'Uploaded: ' + uploadList.join(', '));
+    },
+
+    resetUpload: function () {
+        var uploadList = this.uploadedCsvFiles;
+
+        // Delete uploaded files
+        _.each(uploadList, function (name) {
+            var items = new girder.collections.ItemCollection();
+            items.once('g:changed', function () {
+                if (!items.isEmpty()) {
+                    var item = items.first();
+                    item.destroy();
+
+                    while (uploadList.length) {
+                        uploadList.pop();
+                    }
+                    this.updateUploadWidget();
+                }
+            }, this).fetch({
+                name: name,
+                folderId: this.uploadFolder.id
+            });
+        }, this);
+    },
+
+    validateMetadata: function (save) {
+        // Require dataset to be selected and CSV file to be uploaded
+        var errorMessage = null;
+        if (!this.dataset.id) {
+            errorMessage = 'Please select a dataset.';
+        } else if (_.isEmpty(this.uploadedCsvFiles)) {
+            errorMessage = 'Please upload a CSV file.';
+        }
+
+        if (errorMessage !== null) {
+            isic.showAlertDialog({ text: errorMessage });
+            return;
+        }
+
+        this.dataset.off().on('isic:validated', function (resp) {
```

Comment:
I wonder if this pattern should be retired in favor of `listenTo()`? Even if `once()` is used here, there are two event handlers: one for success and one for error. At most one of those will occur. Does the remaining handler cause the model to leak?